src/coreclr/vm/peimage.cpp

```diff
@@ -1045,21 +1047,24 @@ PTR_PEImageLayout PEImage::CreateLayoutMapped()
         PEImageLayoutHolder pLayout(PEImageLayout::Map(this));
 
         bool fMarkAnyCpuImageAsLoaded = false;
+
         // Avoid mapping another image if we can. We can only do this for IL-ONLY images
-        // since LoadLibrary is needed if we are to actually load code
+        // since LoadLibrary is needed if we are to actually load code (e.g. IJW).
         if (pLayout->HasCorHeader())
         {
-            if (pLayout->IsILOnly())
+            // IJW images must be successfully loaded by the OS to handle
+            // native dependencies, therefore they cannot be mapped.
+            if (!pLayout->IsILOnly())
             {
-                // For CoreCLR, IL only images will always be mapped. We also dont bother doing the conversion of PE header on 64bit,
-                // as done below for the desktop case, as there is no appcompat burden for CoreCLR on 64bit to have that conversion done.
-                fMarkAnyCpuImageAsLoaded = true;
-            }
-            else
-            {
-                // IJW images must be loaded, not mapped
-                ThrowHR(COR_E_BADIMAGEFORMAT);
+                // For compat with older CoreCLR versions we will fallback to the
+                // COR_E_BADIMAGEFORMAT error code if a failure wasn't indicated.
+                loadFailure = FAILED(loadFailure) ? loadFailure : COR_E_BADIMAGEFORMAT;
+                EEFileLoadException::Throw(GetPath(), loadFailure);
             }
+
+            // IL only images will always be mapped. We also dont bother doing the conversion of PE header on 64bit,
+            // as done below for the desktop case, as there is no appcompat burden for CoreCLR on 64bit to have that conversion done.
```

Comment:
> conversion of PE header on 64bit, as done below for the desktop case

I don't see what this comment is referring to - is this still applicable?

Reply:
It is because it was done in .NET Framework. I meant to keep part of it to indicate that it was done on .NET Framework.